Add recursive database owner change to postgres module/state

lucacorti opened the pull request:
Add recursive owner change to postgres.db_alter and expose it via
postgres_database.present parameter.

`owner_recurse` defaults to false to avoid breaking compatibility, but
could arguably be a sane default.

lucacorti committed:
Add recursive database owner change to postgres module/state

Add recursive owner change to postgres.db_alter and expose it via
postgres_database.present parameter.

`owner_recurse` defaults to false to avoid breaking compatibility, but
could arguably be a sane default.

jfindlay commented:
@lucacorti, thanks for the pull request.  There is also a test that needs to be updated.

lucacorti committed:
Fix postgres.db_alter test, add test for owner_recurse=True

The old db_alter test was probably failing anyway. The new test tests
owner_to is called when owner_recurse is set to true.

lucacorti committed:
Merge branch 'develop' of https://github.com/saltstack/salt into develop
